Assets/Scripts/SS3D/Systems/Tile/TileMapCreator/TileMapCreator.cs

```diff
@@ -417,103 +538,140 @@ public void OnDropDownChange()
-                TileMapCreatorTab tab = slot.AddComponent<TileMapCreatorTab>();
-                tab.Setup(asset);
-            }
+                TileObjectSo => false,
+                ItemObjectSo => true,
+                _ => _itemPlacement,
+            };
+            _selectedObject = genericObjectSo;
+            ClearGhosts();
+            _buildGhosts.Add(CreateGhost(genericObjectSo.prefab, GetPointedPosition(!_itemPlacement), _direction));
         }
-
-        public void OnInputFieldSelect()
+        
+        /// <summary>
+        /// Change number of columns in asset grid to fit it's width. Elements of the group will take as much width as possible, but won't exceed width of the menu.
+        /// </summary>
+        private void AdjustGridWidth()
         {
-            _inputSystem.ToggleAllActions(false);
+            GridLayoutGroup grid = _contentRoot.GetComponent<GridLayoutGroup>();
+            float cellWidth = grid.cellSize.x;
+            float paddingWidth = grid.spacing.x;
+            float width = _menuRoot.GetComponent<RectTransform>().rect.width;
+            int constraintCount = Convert.ToInt32(Math.Floor(width / (cellWidth + paddingWidth)));
+            if (constraintCount != grid.constraintCount)
+                grid.constraintCount = constraintCount;
         }
-
-        public void OnInputFieldDeselect()
+        #endregion
+        
+        #region Map
+        private void ClearGrid()
         {
-            _inputSystem.ToggleAllActions(true);
+            for (int i = 0; i < _contentRoot.transform.childCount; i++)
+            {
+                _contentRoot.transform.GetChild(i).gameObject.Dispose(true);
+            }
+        }
+        [Server]
+        public void LoadMap()
+        {
+            if (IsServer)
+            {
+                _tileSystem.Load();
+            }
+            else
+            {
+                Log.Information(this, "Cannot load the map on a client");
+            }
         }
 
-        [ServerOrClient]
-        public void OnPointerEnter(PointerEventData eventData)
+        [Server]
+        public void SaveMap()
         {
-            _mouseOverUI = true;
-            Subsystems.Get<InputSystem>().ToggleBinding("<Mouse>/scroll/y", false);
+            if (IsServer)
+            {
+                _tileSystem.Save();
+            }
+            else
+            {
+                Log.Information(this, "Cannot save the map on a client");
+            }
         }
+        #endregion
 
-        [ServerOrClient]
-        public void OnPointerExit(PointerEventData eventData)
+        /// <summary>
+        /// Get position on the tile grid, that mouse points to.
+        /// </summary>
+        /// <param name="isTilePosition">If true, position snaps to the center of a tile</param>
+        /// <returns></returns>
+        private Vector3 GetPointedPosition(bool isTilePosition = false)
```

Comment:
I feel like this could be useful elsewhere too. I'd say this one belong to the tilehelper class